materialize-snowflake/staged_file.go

```diff
@@ -169,16 +171,19 @@ func (f *stagedFile) encodeRow(row []interface{}) error {
 	return nil
 }
 
-func (f *stagedFile) flush() error {
+func (f *stagedFile) remoteFilePath(file string) string {
+	return filepath.Join(f.uuid, file)
+}
+
+func (f *stagedFile) flush() (string, error) {
 	if err := f.putFile(); err != nil {
-		return fmt.Errorf("flush putFile: %w", err)
+		return "", fmt.Errorf("flush putFile: %w", err)
 	}
-
 	close(f.putFiles)
 	f.started = false
 
 	// Wait for all outstanding PUT requests to complete.
-	return f.group.Wait()
+	return f.uuid, f.group.Wait()
```

Comment:
Where the `f.uuid` return value is used, it looks like we're always doing a `fmt.Sprintf("@flow_v1/%s", dir)` with it. Perhaps instead we could return `fmt.Sprintf("@flow_v1/%s", f.uuid)` here. The return value for this function would then be the "full" staged file(s) directory.

Reply:
@williamhbaker ended up updating some of the templates so in all cases we could feed `@flow_v1` and now we can return `@flow_v1`, previously we had `@flow_v1` as part of the template, hence the formatting of the result in some places, but not others